### PR TITLE
Adding documentation for supporting sap-commerce-cloud-user-id in a Commerce Cloud-backed SPA instance

### PR DESCRIPTION
**Pull request by justinlee01**

### CODE REVIEW

File: _pages/install/backend/installing-sap-commerce-cloud-2105.md [hunk 1]
```diff
@@ -187,6 +187,16 @@ To be able to register users and check out, SAP Commerce Cloud must be configure
...
+### Assisted Service Module
+
+If your storefront is using the Assisted Service Module (ASM), you will need to allow a specific header for customer emulation to fully work.
+
+1. Add the value `sap-commerce-cloud-user-id` to the values of the key `corsfilter.commercewebservices.allowedHeaders` in a properties file.
+
+2. In a `FeaturesConfig` provider, set a key `enableCommerceCloudUserIdHeader` with the value `true`.
+
+By configuring Spartacus in such a way, the storefront will send requests with the header `sap-commerce-cloud-user-id` with the emulated user's ID as its value. This is to prevent ambiguity errors when making requests to the OCC.
+
```
Comment (plabadie) [inline _pages/install/backend/installing-sap-commerce-cloud-2105.md:199]: Could it be relevant to host this doc on the ASM page?  There is even a section about CORS in there already `_pages/dev/features/asm.md`

File: _pages/install/backend/installing-sap-commerce-cloud-2105.md [hunk 2]
```diff
@@ -187,6 +187,16 @@ To be able to register users and check out, SAP Commerce Cloud must be configure
 - For information on Spartacus and CORS settings, see [{% assign linkedpage = site.pages | where: "name", "cors.md" %}{{ linkedpage[0].title }}]({{ site.baseurl }}{% link _pages/install/cors.md %}).
 - There are a number of other Spartacus features that also require additional CORS settings. For more information about CORS, see [ycommercewebservices Extension](https://help.sap.com/viewer/9d346683b0084da2938be8a285c0c27a/latest/en-US/8c91f3a486691014b085fb11c44412ff.html) in the SAP Help Portal.
 
+### Assisted Service Module
+
+If your storefront is using the Assisted Service Module (ASM), you will need to allow a specific header for customer emulation to fully work.
+
+1. Add the value `sap-commerce-cloud-user-id` to the values of the key `corsfilter.commercewebservices.allowedHeaders` in a properties file.
+
+2. In a `FeaturesConfig` provider, set a key `enableCommerceCloudUserIdHeader` with the value `true`.
```
Comment (plabadie) [inline _pages/install/backend/installing-sap-commerce-cloud-2105.md:196]: Suggestion: for this part an example might be useful.

File: _pages/dev/features/asm.md [hunk 3]
```diff
@@ -112,9 +112,30 @@ For development purposes only, you can set the value to a wildcard (`*`), as sho
 ```text
 corsfilter.assistedservicewebservices.allowedOrigins=*
 ```
-
 **Note:** This wildcard configuration is flexible for development environments, but it is not secure. A more restrictive configuration is required for production use.
 
+### Additional CORS Configuration for Customer Emulation
+
+In order for ASM customer emulation to work, you need to allow a specific header. You must add `sap-commerce-cloud-user-id` to the values of the key `corsfilter.commercewebservices.allowedHeaders` in a properties file. The following is an example:
```
Comment (justinlee01) [inline _pages/dev/features/asm.md:119]:
```suggestion
In order for ASM customer emulation to work, you need to allow a specific header. You must add `sap-commerce-cloud-user-id` to the values of the key `corsfilter.commercewebservices.allowedHeaders` in the CX commerce configuration. The following is an example:
```

File: _pages/dev/features/asm.md [hunk 4]
```diff
@@ -112,9 +112,30 @@ For development purposes only, you can set the value to a wildcard (`*`), as sho
 ```text
 corsfilter.assistedservicewebservices.allowedOrigins=*
 ```
-
```
Comment (justinlee01) [inline _pages/dev/features/asm.md:115]: Revert this.

**Comment by gladius-mtl** [comment]
This content was originally intended only for the 5.1 release, but it now needs to be published as part of a backport to 4.3.x as well, so a new PR has been created that supersedes this one: https://github.com/SAP/spartacus-docs/pull/1515

**Pull request closed without merging**